test/select_tests.cpp

```diff
@@ -805,3 +805,123 @@ TEST(IntervalLiteral) {
     }
   }
 }
+TEST(LockingClauseWithoutWaitPolicy) {
+  SelectStatement* stmt;
+  TEST_PARSE_SQL_QUERY(
+      "SELECT * FROM t WHERE a = 10 FOR UPDATE;"
+      "SELECT * FROM t WHERE a = 10 FOR SHARE;"
+      "SELECT * FROM t WHERE a = 10 FOR NO KEY UPDATE;"
+      "SELECT * FROM t WHERE a = 10 FOR KEY SHARE;",
+      result, 4);
+
+  stmt = (SelectStatement*)result.getStatement(0);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForUpdate);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::None);
+
+  stmt = (SelectStatement*)result.getStatement(1);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForShare);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::None);
+
+  stmt = (SelectStatement*)result.getStatement(2);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForNoKeyUpdate);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::None);
+
+  stmt = (SelectStatement*)result.getStatement(3);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForKeyShare);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::None);
+}
+
+TEST(LockingClauseWithWaitPolicy) {
```

Comment:
Please also check
- the number of elements in `lockings`
- `tables` (of the struct `LockingClause`)